### DISCUSSION TITLE
Rich text editor for next js

### DISCUSSION THREAD

**Discussion by birshu123** [vercel/next.js#11423] [Help]
Dear Community Members,

Please suggest some rich text editor for Next js. I am developing a cms where i need to provide images and text design in my content. Thank You.

**Comment by isBatak** [comment]
you can check slate https://www.slatejs.org/examples/richtext I used it in one project and it turned out to be OK :D

**Reply by birshu123** [reply]
Dear isBatak, i checked it, i have not seen image upload option in this editor.

**Reply by sgc0014** [reply]
so did you find any good editor? I am also in need of it but can't find one for nextjs.

**Reply by louisjuliendo** [reply]
Try [draft.js](https://draftjs.org/). It works well with Next

**Reply by smilegupta** [reply]
You can try react-quill. Just make sure you alter the way of importing react-quill to something like this import dynamic from "next/dynamic";
const ReactQuill = dynamic(() => import("react-quill"), { ssr: false });

**Reply by nitink66** [reply]
But the problem with reat-quill in nextjs is it doesn't register inline styles, and image resizing problems,blob too won't get register

**Reply by EliasTouil** [reply]
> But the problem with reat-quill in nextjs is it doesn't register inline styles, and image resizing problems,blob too won't get register

I'm here because I'm experiencing problems with Quill too, it integrated nicely for basic functionality, but as soon as I wanted to get some control over it it became impossible.

**Reply by Piyusharora2003** [reply]
> const ReactQuill = dynamic(() => import("react-quill"), { ssr: false });

Thanks it helps a lot..

**Reply by SriSomanaath** [reply]
"use client";

import React, { useState } from 'react';
import dynamic from 'next/dynamic';
import styles from './page.module.css';

const SunEditor = dynamic(() => import('suneditor-react'), { ssr: false });
import 'suneditor/dist/css/suneditor.min.css';

const Page = () => {
  const [content, setContent] = useState('');

  const handleEditorChange = (content: string) => {
    setContent(content);
    console.log('Editor Content:', content);
  };

  return (
    <div className={styles.container}>
      <SunEditor
        setOptions={{
          buttonList: [
            ['undo', 'redo', 'font', 'fontSize', 'formatBlock'],
            ['bold', 'underline', 'italic', 'strike', 'subscript', 'superscript'],
            ['removeFormat'],
            '/',
            ['fontColor', 'hiliteColor'],
            ['outdent', 'indent'],
            ['align', 'horizontalRule', 'list', 'table'],
            ['link', 'image', 'video'],
            ['fullScreen', 'showBlocks', 'codeView'],
          ],
        }}
        onChange={handleEditorChange}
        height="400px"
      />
      <div className={styles.preview}>
        <div dangerouslySetInnerHTML={{ __html: content }}></div>
      </div>
    </div>
  );
};

export default Page;



SunEditor is the best. Here is the code you can use

**Comment by Igor2122** [comment]
did anyone use mui-rte?

**Reply by nitink66** [reply]
@Igor2122  is it possible to use mui-rte along with next js ?

**Reply by klawingco** [reply]
I tried using it on our project, and I am stuck with the initial value not rendering right.

**Comment by karthik-js** [comment]
Does anyone have ideas which library kind of work and give us html string with inline styles.

**Reply by nitink66** [reply]
If you are using nextjs ,then i recommend sunEditor

**Reply by karthik-js** [reply]
@nitink66 Will it return html with inline styles? Currently I am using react-quill, it doesn't return html with inline styles.

**Reply by nitink66** [reply]
Yup works perfectly!, also has built in image resize ability, so you don't need to again depend on other libraries like [image-resize-module](https://www.npmjs.com/package/quill-image-resize-module) as for react-quill.
I also use [HTMR package](https://www.npmjs.com/package/htmr) 
Also I suggest you to go with [SunEditor-React](https://github.com/mkhstar/suneditor-react/) for nextjs 
[Implement Suneditor in React](https://aawara.me/blogs/how-to-implement-rich-text-editor-in-nextjs)

**Reply by seyedmahdii** [reply]
> Yup works perfectly!, also has built in image resize ability, so you don't need to again depend on other libraries like [image-resize-module](https://www.npmjs.com/package/quill-image-resize-module) as for react-quill. I also use [HTMR package](https://www.npmjs.com/package/htmr) Also I suggest you to go with [SunEditor-React](https://github.com/mkhstar/suneditor-react/) for nextjs [Implement Suneditor in React](https://aawara.me/blogs/how-to-implement-rich-text-editor-in-nextjs)

How do you add "iframe" to the text editor using suneditor?

**Reply by Aalian2241** [reply]
YOU ARE THE GOAT ISTG

**Comment by Ksondzyk** [comment]
srt

**Comment by Ksondzyk** [comment]
rtrt

**Comment by gomflo** [comment]
what about https://www.tiptap.dev/?

**Reply by seyedmahdii** [reply]
Does it support "iframe"? I didn't find it in the docs.

**Comment by Gabsys** [comment]
Bump for this. Anyone have a good suggestion? SunEditor & react-draft-wysiwyg were giving me issue with content change (They automatically scroll to top upon content change), CKEditor can't have alignments plugin :/

**Reply by Aniket-Jena-02** [reply]
Did you find any?

**Comment by eughenio-rhevolut** [comment]
Any news about this topic???

**Comment by abdulkadirkg** [comment]
I am using CKEditor (for markdown, I am using [Turndown](https://www.npmjs.com/package/turndown)). It supports image upload and table features. If you want to headerless table you can check [this](https://www.npmjs.com/package/@joplin/turndown-plugin-gfm) also.

**Comment by Jafoor** [comment]
dropdown not working in next.js-13.4.5, for react-draft-wysiwyg. I also asked [question ](https://stackoverflow.com/questions/76536348/any-dropdown-options-not-working-with-react-draftjs-wysiwyg-and-next-js-typescri) on stackoverflow.

**Comment by rinald-ta** [comment]
How about Lexical? https://lexical.dev/
It's actually a text editor framework, so you'd actually need to create the text editor yourself. It's also officially maintained by Meta/Facebook, so it works great with React (and Next.js too, probably).

You can take a look at the playground here to see what's possible with Lexical: https://playground.lexical.dev/

**Comment by Balde17** [comment]
hello w**_orld_**

**Comment by Mina-Massoud** [comment]
Hey everyone! 👋
I’ve just released a Rich Text Editor built with shadcn/ui and Tailwind CSS.

I made it because I was tired of complex or outdated editors — so here’s a clean, modern, and open-source alternative you can actually use in your projects.

👉 Live demo: [mina-rich-editor.vercel.app](https://mina-rich-editor.vercel.app/)

💻 GitHub repo: [github.com/Mina-Massoud/Mina-Rich-Editor](https://github.com/Mina-Massoud/Mina-Rich-Editor)

Would love to hear your feedback or suggestions!

**Comment by abhishekmardiya** [comment]
[next-mdx-remote ](https://github.com/hashicorp/next-mdx-remote) is great and works seamlessly with Next.js.

You can see it in action in my implementation here:
[GitHub – async-awaits/components/editor](https://github.com/abhishekmardiya/async-awaits/tree/main/components/editor)
